frontend/public/components/chargeback.tsx

```diff
@@ -174,42 +173,42 @@ const DataCell = ({name, value}) => {
   return value;
 };
 
-const DataTable = ({rows, orderBy, sortBy, applySort, keys}:DataTableProps) => {
+const DataTable = ({cols, rows}:DataTableProps) => {
   const size = _.clamp(Math.floor(12 / _.size(rows[0])), 1, 4);
   const className = `col-md-${size}`;
-  return <div className="co-m-table-grid co-m-table-grid--bordered" style={{marginTop: 20, marginLeft: -15, marginRight: -15}}>
-    <ListHeader>
-      {_.map(keys, k => <ColHead
-        className={classNames(className, {'text-right': _.isFinite(_.get(rows, [0, k]))})}
-        key={k}
-        sortField={k}
-        sortFunc={k}
-        currentSortOrder={orderBy}
-        currentSortField={sortBy}
-        currentSortFunc={sortBy}
-        applySort={applySort}>
-        {k.replace(/_/g, ' ')}
-      </ColHead>
-      )}
-    </ListHeader>
-    <div className="co-m-table-grid__body">
-      {_.map(rows, (r, i) => <div className="row co-resource-list__item" key={i}>
-        {_.map(r, (v, k) => <div className={className} key={k}><DataCell name={k} value={v} /></div>)}
-      </div>)}
-    </div>
-  </div>;
+
+  const DataTableHeader = () => _.map(cols, k => {
+    return ({
+      title: k.replace(/_/g, ' '),
+      sortField: k,
+      transforms: [sortable],
+      props: { className },
+    });
+  });
+
+  const DataTableRows = ({componentProps: {data}}) => _.map(data, (r) => _.map(r, (v, k) => {
+    return {title: <DataCell name={k} value={v} />};
+  }));
+
+  return <Table aria-label="Usage Report"
+    Header={DataTableHeader}
+    Rows={DataTableRows}
+    data={rows}
+    virtualize={false}
+    loaded={true} />;
 };
 
 class ReportData extends React.Component<ReportDataProps, ReportDataState> {
   constructor(props) {
     super(props);
     this.state = {
       inFlight: false,
+      isFinished: false,
```

Comment:
Can you clarify the difference between `inFlight` and `isFinished`?

Reply:
inFlight is when the api request has been made.  isFinished is when the report get's back a condition of running and false, meaning the Report has truly finished.

Reply:
I would change the variable name to `isReportFinished` to make it clear that it's not referring to the request being finished. It's probably worth a comment as well.

Reply:
ok, renamed.